stackit/internal/services/observability/alertgroup/resource.go

```diff
@@ -0,0 +1,566 @@
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectId := model.ProjectId.ValueString()
+	instanceId := model.InstanceId.ValueString()
+	alertGroupName := model.Name.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "alert_group_name", alertGroupName)
+	ctx = tflog.SetField(ctx, "instance_id", instanceId)
+
+	_, err := a.client.DeleteAlertgroup(ctx, alertGroupName, instanceId, projectId).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting alert group", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+
+	tflog.Info(ctx, "Alert group deleted")
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: project_id,instance_id,name
+func (a *alertGroupResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, core.Separator)
+
+	if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing scrape config",
+			fmt.Sprintf("Expected import identifier with format: [project_id],[instance_id],[name]  Got: %q", req.ID),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("instance_id"), idParts[1])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("name"), idParts[2])...)
+	tflog.Info(ctx, "Observability alert group state imported")
+}
+
+// toCreatePayload generates the payload to create a new alert group.
+func toCreatePayload(ctx context.Context, model *Model) (*observability.CreateAlertgroupsPayload, error) {
+	if model == nil {
+		return nil, fmt.Errorf("nil model")
+	}
+
+	payload := observability.CreateAlertgroupsPayload{}
+
+	if !utils.IsUndefined(model.Name) {
+		payload.Name = model.Name.ValueStringPointer()
+	}
+
+	if !utils.IsUndefined(model.Interval) {
+		payload.Interval = model.Interval.ValueStringPointer()
+	}
+
+	if !utils.IsUndefined(model.Rules) {
+		rules, err := toRulesPayload(ctx, model)
+		if err != nil {
+			return nil, err
+		}
+		payload.Rules = &rules
+	}
+
+	return &payload, nil
+}
+
+// toRulesPayload generates rules for create payload.
+func toRulesPayload(ctx context.Context, model *Model) ([]observability.UpdateAlertgroupsRequestInnerRulesInner, error) {
+	if model.Rules.Elements() == nil || len(model.Rules.Elements()) == 0 {
+		return []observability.UpdateAlertgroupsRequestInnerRulesInner{}, nil
+	}
+
+	var rules []rule
+	diags := model.Rules.ElementsAs(ctx, &rules, false)
+	if diags.HasError() {
+		return nil, core.DiagsToError(diags)
+	}
+
+	var oarrs []observability.UpdateAlertgroupsRequestInnerRulesInner
+	for i := range rules {
+		rule := rules[i]
+		oarr := observability.UpdateAlertgroupsRequestInnerRulesInner{}
```

Comment:
I keep seeing this pattern in the codebase, personally I would prefer the more ideomatic approach:

```suggestion
	for i, rule := range rules {
		oarr := observability.UpdateAlertgroupsRequestInnerRulesInner{}
```